layer/src/conv.rs

```diff
@@ -0,0 +1,351 @@
+// TODO: refactor looped assignments as iterator collections
+
+use displaydoc::Display;
+use ndarray::{s, Array1, Array2, Array3, ArrayBase, Data, ErrorKind, Ix2, Ix3, ShapeError};
+use thiserror::Error;
+
+/// A 1-dimensional convolutional layer.
+pub struct Conv1D {
+    // params
+    weights: Array2<f32>,
+    bias: Option<Array3<f32>>,
```

Comment:
```suggestion
    weights: Array3<f32>,
    bias: Option<Array1<f32>>,
```

I still need to double check it but bias being 3-D and with that higher dimensional then weights seems strange, tbh.

Reply:
the weights are already reshaped into the dimension needed during the run, such that we don't have to reshape them every time.

the bias is just extended with singleton dimensions from `(bias_size,)` to `(1, bias_size, 1)` for the same reason.

Reply:
Oh, you are right I even preciously checked that the transformation was correct :see_no_evil: